### ISSUE TITLE
User story no. 1 after hydrological expert group discussion

### ISSUE THREAD

**Issue by rcapell** [switchonproject/sip-html5#13]
As a hydrologist, I need to find a list of data links as provided in Emmas data table by hierarchical keywords as provided in CUAHSI's tabular ontology table in column B to D, so I can efficiently identify precise and required data for my study.

CUAHSI ontology: http://his.cuahsi.org/documents/Tabular_Ontology.xlsx
Emma Metadata: already in sip

**Comment by schlob** [comment]
the use of semantic technologies is out of scope atm what we are aiming at in WP2 is to develop a simple HTML5 client for the SIP backend supporting Step 2 discovery and download.

The use of ontolgies implies the use of a reasoner and possibly a completely different information model and technology. At least for cismet this 100% out of scope (cannot make a statement for deltares though)

**Comment by schlob** [comment]
But mabe I am not 100% getting what you actually want to achieve:-)

**Comment by esafa** [comment]
Nobody was asking for a "semantic reasoner". Semantic technology is absolutely not needed for this user story. If they would have asked for that I (Esa) would have said "out of scope" and "epic" but they did not ask for that.

Lets have a look at what they need (in more technical terms): 
- keyword search based on three fixed predetermined lists of keywords. 

If you wanted, you could support this without any change to the SIM at all (using namespace-constructs). After all, SIM already supoprts keyword search!  The only problem is that freetext keywords are good for understanding the dataset but fairly useless for search. So, in technical terms, we need... 

(1) summerworker to add the CUASHI keywords to the backend database

(2) limit the keyword search to the keywords in columns B, D, and C of the excel list (possibly some kind of pulldown menu? idk?).

This seems like a task that would take a few days to implement.

**Comment by p-a-s-c-a-l** [comment]
Yes, please see my reply to reply to [User story René 01](https://github.com/switchonproject/sip-html5/issues/4#issuecomment-70230054).

It applies also to this user story. It's mainly a matter of tagging the available resources according to the CUAHSI ontology (no problem with Expert GUI) and how to select the keywords in the HTML GUI.

BTW, just columns B to **D**? or B to **C**?

**Comment by schlob** [comment]
ok :relieved:

for a relatively fix content this works, but this means adding everything, explicitly and  "all the time" on the data side to get the expected results. Doesn't it?

**Comment by esafa** [comment]
Yes in a perfect world we would need to add the three levels of keywords to
each and every dataset added and stored in the SIM. However, Rene also
suggested that we would allow some datasets to only have the coarse topic
(column B) and skip the detail classification, whereas other (most) dataset
would have more detailed classification with all columns: B, C and D.

Pragmatic me thinks.
//eF



On Wed, Jan 21, 2015 at 4:49 PM, Sascha Schlobinski <
notifications@github.com> wrote:

> ok [image: :relieved:]
>
> for a relatively fix content this works, but this means adding everything,
> explicitly and "all the time" on the data side to get the expected results.
> Doesn't it?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/switchonproject/sip-html5/issues/13#issuecomment-70861316>
> .
>



-- 
Esa Falkenroth
mobile: 070-666-14-14 (intl +46706661414)
work: 011-4958298 (intl +46114958298)
skype: Esa.Falkenroth
email: nospam@falkenroth.se (for business mail)
email: esa@falkenroth.se (for private mail)

**Comment by rcapell** [comment]
@Pascal
we meant Columns B (Nav1), C (Search1), D (Search2)

**Comment by schlob** [comment]
we will provide a mockup here asap

**Comment by esafa** [comment]
Great. Duro was most interested in the UI-issues. Would it be possible to
get a png of the mockup... or even google doc. Or should we install
balsamique?



On Wed, Jan 21, 2015 at 5:08 PM, Sascha Schlobinski <
notifications@github.com> wrote:

> we will provide a mockup here asap
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/switchonproject/sip-html5/issues/13#issuecomment-70864973>
> .
>



-- 
Esa Falkenroth
mobile: 070-666-14-14 (intl +46706661414)
work: 011-4958298 (intl +46114958298)
skype: Esa.Falkenroth
email: nospam@falkenroth.se (for business mail)
email: esa@falkenroth.se (for private mail)

**Comment by schlob** [comment]
lets start with pngs probably this will be ok as a basis for communication

**Comment by esafa** [comment]
pngs are great.

Is there anything unclear remaining with this userstory? If there is, please post your questions here (Gerben mentioned that Martin still was waiting for the discussion to "finish" so there may be unclear things that need to be answered).

**Comment by schlob** [comment]
pascal will provide a mockup for this user story today.

OFC this will be an iterative process - won't it?

**Comment by esafa** [comment]
Yes... ofc.. iterative process: User story group agreed to follow these
simple principles "I-N-V-E-S-T":

Independent (self-contained) user story, with no inherent dependency on
another user story.

Negotiable: userstory negociated and rewritten until the
userstory-group+cismet agree.

Valuable: A user story must most valuable to an end user doing
SEARCH/FIND/DOWNLOAD

Estimatable: Sufficiently clear user story to enable estimate of size
(tshirt sizes: S/M/L)

Small sized: User stories sufficiently small/fast so we have time to do
several stories before the GA.
Testable: The user story (plus info on github) must contain sufficient
information to make it possible to “test”/determine if the userstory was
successfully implemented.

**Comment by p-a-s-c-a-l** [comment]
Thanks @rcapell !

I've extracted the list of hierarchical keywords from the tabular ontology:
https://github.com/switchonproject/cids-custom-switchon-server/wiki/Tabular-CUASHI-Ontology
Please check,if this what you expect.

> @Pascal we meant Columns B (Nav1), C (Search1), D (Search2)

**Comment by p-a-s-c-a-l** [comment]
**UI Mockus for User Story 1:**

![uc010-010-cuahsi-ontology](https://cloud.githubusercontent.com/assets/1788226/5876005/c1bb5af4-a313-11e4-8a32-a6bdf5a52044.png)

![uc010-020-cuahsi-ontology-toplevel](https://cloud.githubusercontent.com/assets/1788226/5876007/c4fed3d0-a313-11e4-8743-fb21a42ed4ac.png)

![uc010-030-cuahsi-ontology-toplevel-select](https://cloud.githubusercontent.com/assets/1788226/5876008/c71611ec-a313-11e4-80fe-06e9ac6adad4.png)

![uc010-040-cuahsi-ontology-sublevel](https://cloud.githubusercontent.com/assets/1788226/5876010/c8d9b128-a313-11e4-9316-ed027ce02df0.png)

Addendum to last mockup: The search in the last example would **not** return any "inorganic" or "Nutrient" resources. See this [table](https://github.com/switchonproject/cids-custom-switchon-server/wiki/Tabular-CUASHI-Ontology) for the keyword hierarchy of the CUASHI Ontology.

**Comment by p-a-s-c-a-l** [comment]
**Rapid Protype Mockup in HTML GUI:**

![mockup-demo](https://cloud.githubusercontent.com/assets/1788226/5876508/0b070ce0-a318-11e4-93a8-6a3dcc978941.gif)

This is how it could look like in the HTML5 GUI. Note that there is even no additional click in a list required (step no. 2 in Balsamiq Mockups above)!

**Comment by esafa** [comment]
Great. I will leave the discussion on the mockup to @rcapell, @rcnijzink and duro.  However, I doubt  the hydrologists will remember the CUASHI keywords well enough to even know what to type in a search box. You may consider pulldowns since this is a fixed list.

@Remko, Rene, Duro, please have a response as soon as possible.

Cheers,//eF

**Comment by esafa** [comment]
@p-a-s-c-a-l  the list looks identical to the excel sheet with only three odd things... All thirdlevel keywords have been modified to remove the redundant 2nd level keyword (separated by comma). This is done for all entries except three:

Area	Area, ice
Area, atmosphere
Optical, water

**Comment by duropa** [comment]
it is nice...i would say that most of the hydrologists are not aware of some cuashi list so fixed list is a good suggestion (but i do not know what other options are possible...
cheers
d.

**Comment by schlob** [comment]
mmh :-) here you need to know: something with chemical.... and you get all the fitting keywords where in a fix list you might need to scroll to somewhere (and if you don'tknow chemical you are not better off either.
Can we really presume that a hydrologist whants use a keyword list he has no clue of?
For me 2 steps would make sense (1) hydrologist has an idea what he is looking for but does not know the keywords exactly - he uses the UI as proposed by Pascal (2.1) he finds a fitting keyword -->good(2.2) he does not and needs to browse all to get an idea.
So maybe it is a good idea to combine both: first show all and then filter rather than show only what fits to the substring.

just my 2 cents

**Comment by esafa** [comment]
haha... I was expecting this. .. all computer scientists jump in and join
the discussion merely minutes after the draft mockup was posted.

Ok. here is my five cents (for what it is worth). All CUASHI keywords
actually fit on a normal screen if properly organised. Just make tick-boxes
where the scientist can make choice(s). One click per "class". That would
be much less work than trying to find "muddy water" by first pressing "A"
and the getting all the choices.. and since none of those match what he/she
wants he/she  deletes the "A" and types "B"  to get more choices. and
finally after 31 keystokes finding "Optical water" (which matches).

But... in the end... it is rene, remko, duro that can say if they want to
type the the keyword(s) in or if they prefer selecting the keyword(s) from
a list.  :)

//eF




On Fri, Jan 23, 2015 at 5:04 PM, Sascha Schlobinski <
notifications@github.com> wrote:

> mmh :-) here you need to know: something with chemical.... and you get all
> the fitting keywords where in a fix list you might need to scroll to
> somewhere (and if you don'tknow chemical you are not better off either.
> Can we really presume that a hydrologist whants use a keyword list he has
> no clue of?
> For me 2 steps would make sense (1) hydrologist has an idea what he is
> looking for but does not know the keywords exactly - he uses the UI as
> proposed by Pascal (2.1) he finds a fitting keyword -->good(2.2) he does
> not and needs to browse all to get an idea.
> So maybe it is a good idea to combine both: first show all and then filter
> rather than show only what fits to the substring.
>
> just my 2 cents
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/switchonproject/sip-html5/issues/13#issuecomment-71215647>
> .
>



-- 
Esa Falkenroth
mobile: 070-666-14-14 (intl +46706661414)
work: 011-4958298 (intl +46114958298)
skype: Esa.Falkenroth
email: nospam@falkenroth.se (for business mail)
email: esa@falkenroth.se (for private mail)

**Comment by esafa** [comment]
I think I just said what Sascha said but in another way?

On Fri, Jan 23, 2015 at 5:19 PM, Dr Falkenroth <nospam@falkenroth.se> wrote:

> haha... I was expecting this. .. all computer scientists jump in and join
> the discussion merely minutes after the draft mockup was posted.
>
> Ok. here is my five cents (for what it is worth). All CUASHI keywords
> actually fit on a normal screen if properly organised. Just make tick-boxes
> where the scientist can make choice(s). One click per "class". That would
> be much less work than trying to find "muddy water" by first pressing "A"
> and the getting all the choices.. and since none of those match what he/she
> wants he/she  deletes the "A" and types "B"  to get more choices. and
> finally after 31 keystokes finding "Optical water" (which matches).
>
> But... in the end... it is rene, remko, duro that can say if they want to
> type the the keyword(s) in or if they prefer selecting the keyword(s) from
> a list.  :)
>
> //eF
>
>
>
>
> On Fri, Jan 23, 2015 at 5:04 PM, Sascha Schlobinski <
> notifications@github.com> wrote:
>
>> mmh :-) here you need to know: something with chemical.... and you get
>> all the fitting keywords where in a fix list you might need to scroll to
>> somewhere (and if you don'tknow chemical you are not better off either.
>> Can we really presume that a hydrologist whants use a keyword list he has
>> no clue of?
>> For me 2 steps would make sense (1) hydrologist has an idea what he is
>> looking for but does not know the keywords exactly - he uses the UI as
>> proposed by Pascal (2.1) he finds a fitting keyword -->good(2.2) he does
>> not and needs to browse all to get an idea.
>> So maybe it is a good idea to combine both: first show all and then
>> filter rather than show only what fits to the substring.
>>
>> just my 2 cents
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/switchonproject/sip-html5/issues/13#issuecomment-71215647>
>> .
>>
>
>
>
> --
> Esa Falkenroth
> mobile: 070-666-14-14 (intl +46706661414)
> work: 011-4958298 (intl +46114958298)
> skype: Esa.Falkenroth
> email: nospam@falkenroth.se (for business mail)
> email: esa@falkenroth.se (for private mail)
>



-- 
Esa Falkenroth
mobile: 070-666-14-14 (intl +46706661414)
work: 011-4958298 (intl +46114958298)
skype: Esa.Falkenroth
email: nospam@falkenroth.se (for business mail)
email: esa@falkenroth.se (for private mail)

**Comment by schlob** [comment]
:metal:

**Comment by rcnijzink** [comment]
Sorry for the late reply, but to comment on this: i think i agree with Duro and Esa. Not every hydrologist will know the cuashi keywords by heart, and scrolling through a list with keywords may be easier in that case

**Comment by p-a-s-c-a-l** [comment]
Ok, here is an updated Mockup.

**CUASHI Keywords Selection List**

![cuashi-list](https://cloud.githubusercontent.com/assets/1788226/5900299/ac797b8e-a567-11e4-891a-d4956d40cd81.gif)

**Comment by esafa** [comment]
I noticed the mockup keywords contains a mix of full and partial CUASHI
terms.

*  "Level" is a CUASHI term and matches an entry in column "C" row 2
(Search1[2])  but..
*  "Ice" doesnt match any entry in CUASHI in column B, C nor D. The word
"ice" does appear as PART of several CUASHI keywords such as "Level,Ice"
(col D, row 2) and "Area,Ice" (col D, row14)...

My question is... do we want to use the CUASHI terms in column D
"Search2[3]" strictly, or do we want to make our OWN keyword list. My
experience is that new keyword lists take time to agree on,..  and they
also need to be maintained and modified lists may make it harder to
cooperate with CUASHI in the future. The benefit of deviating from CUASHI
is that we can add our own keywords later.

Mostly an issue of flexibility and development time vs compability. Make
sure you decide wisely.




On Mon, Jan 26, 2015 at 2:32 PM, Pascal Dihé <notifications@github.com>
wrote:

> Ok, here is an updated Mockup.
>
> *CUASHI Keywords Selection List*
>
> [image: cuashi-list]
> <https://cloud.githubusercontent.com/assets/1788226/5900299/ac797b8e-a567-11e4-891a-d4956d40cd81.gif>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/switchonproject/sip-html5/issues/13#issuecomment-71460682>
> .
>



-- 
Esa Falkenroth
mobile: 070-666-14-14 (intl +46706661414)
work: 011-4958298 (intl +46114958298)
skype: Esa.Falkenroth
email: nospam@falkenroth.se (for business mail)
email: esa@falkenroth.se (for private mail)

**Comment by schlob** [comment]
it should be the correct cuashi list. We can add other keywords as well ofc they could be defined by
us or even come from other taxonomies etc.

**Comment by esafa** [comment]
We expect to hire a summerworker (June-July), to tag all the SIP-resources
with these additional keywords... so by then we must know which one we want
to use.

On Mon, Jan 26, 2015 at 3:31 PM, Sascha Schlobinski <
notifications@github.com> wrote:

> it should be the correct cuashi list. We can add other keywords as well
> ofc they could be defined by
> us or even come from other taxonomies etc.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/switchonproject/sip-html5/issues/13#issuecomment-71468443>
> .
>



-- 
Esa Falkenroth
mobile: 070-666-14-14 (intl +46706661414)
work: 011-4958298 (intl +46114958298)
skype: Esa.Falkenroth
email: nospam@falkenroth.se (for business mail)
email: esa@falkenroth.se (for private mail)

**Comment by p-a-s-c-a-l** [comment]
Atm there are four different keyword lists in the MetaData Repository:
- Inspire Themes 1.0 (fixed taxonomy)
- Inspire Topic Categories (fixed taxonomy)
- Free Keywords (open list, populated with arbitrary keywords from imported resources)
- CUASHI tabular ontology (fixed taxonomy)

Accordingly, there are 4 different keyword list in the UI:

![keyword-lists](https://cloud.githubusercontent.com/assets/1788226/5901288/9e4134b2-a571-11e4-8189-54dc7473a4e6.png)

**Comment by p-a-s-c-a-l** [comment]
> I noticed the mockup keywords contains a mix of full and partial CUASHI terms. * "Level" is a CUASHI term and matches an entry in column "C" row 2 (Search1[2]) but.. * "Ice" doesnt match any entry in CUASHI in column B, C nor D. The word "ice" does appear as PART of several CUASHI keywords such as "Level,Ice" (col D, row 2) and "Area,Ice" (col D, row14)...

OK, I'll check the list.

Btw, we could also integrate the CUASHI tabular ontology list into the Free Keywords List. What do you think?

**Comment by schlob** [comment]
makes sense - we can adapt this later if required

**Comment by esafa** [comment]
I guess free keywords search should span over all text-based columns in
metadata such as "data set abstract text" , name of organisation providing
the data set etc... How you solve this technically is a matter of
choice/preference. Im a database-guy so I know how I would have done it :)


On Tue, Jan 27, 2015 at 1:15 PM, Sascha Schlobinski <
notifications@github.com> wrote:

> makes sense - we can adapt this later if required
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/switchonproject/sip-html5/issues/13#issuecomment-71638903>
> .
>



-- 
Esa Falkenroth
mobile: 070-666-14-14 (intl +46706661414)
work: 011-4958298 (intl +46114958298)
skype: Esa.Falkenroth
email: nospam@falkenroth.se (for business mail)
email: esa@falkenroth.se (for private mail)

**Comment by p-a-s-c-a-l** [comment]
> I guess free keywords search should span over all text-based columns in metadata such as "data set abstract text" , name of organisation providing the data set etc...

This type of free keywords search will be implemented too. It's called "free text search".

However, here we are talking about lists of keywords that are available in the DB and how to present this to the user. As we've learned, users don't know the CUASHI, INSPIRE, etc. keywords by heart. Therefore a free text search (as proposed in the 1st mockup) will not help here. Instead, we have to show a list of keywords to select from.

Atm, the following keywords lists are available:
- [INSPIRE themes 1.0](https://github.com/switchonproject/cids-custom-switchon-server/wiki/INSPIRE-themes-1.0)
- [Tabular CUASHI Ontology](https://github.com/switchonproject/cids-custom-switchon-server/wiki/Tabular-CUASHI-Ontology)
- [Open Keyword List] (https://github.com/switchonproject/cids-custom-switchon-server/wiki/Open-Keyword-List)

Moreover, the _Open Keyword List_ does already contain some of the more general CUASHI  keywords (e.g. "area"). Therefore a search for those CUASHI Keywords will already deliver some results!

Now, the question to @duropa, @rcapell and @rcnijzink  is: Should be show one combined list with INSPIRE, CUASHI and free keywords or 3 different list as shown in [this mockup](https://github.com/switchonproject/sip-html5/issues/13#issuecomment-71470592)?

**Comment by esafa** [comment]
Adding additional keyword lists was not part of the user story and hence it
would be feature creep. Let us not clutter the GUI with things that the
hydrologist have not asked for.


On Tue, Jan 27, 2015 at 2:09 PM, Pascal Dihé <notifications@github.com>
wrote:

> I guess free keywords search should span over all text-based columns in
> metadata such as "data set abstract text" , name of organisation providing
> the data set etc...
>
> This type of free keywords search will be implemented too. It's called
> "free text search".
>
> However, here we are talking about lists of keywords that are available in
> the DB and how to present this to the user. As we've learned, users don't
> know the CUASHI, INSPIRE, etc. keywords by heart. Therefore a free text
> search (as proposed in the 1st mockup) will not help here. Instead, we have
> to show a list of keywords to select from.
>
> Atm, the following keywords lists are available:
>
>    - INSPIRE themes 1.0
>    <https://github.com/switchonproject/cids-custom-switchon-server/wiki/INSPIRE-themes-1.0>
>    - Tabular CUASHI Ontology
>    <https://github.com/switchonproject/cids-custom-switchon-server/wiki/Tabular-CUASHI-Ontology>
>    - Open Keyword List
>    <https://github.com/switchonproject/cids-custom-switchon-server/wiki/Open-Keyword-List>
>
> Moreover, the *Open Keyword List* does already contain some of the more
> general CUASHI keywords (e.g. "area"). Therefore a search for those CUASHI
> Keywords will already deliver some results!
>
> Now, the question to @duropa <https://github.com/duropa>, @rcapell
> <https://github.com/rcapell> and @rcnijzink <https://github.com/rcnijzink>
> is: Should be show one combined list with INSPIRE, CUASHI and free keywords
> or 3 different list as shown in this mockup
> <https://github.com/switchonproject/sip-html5/issues/13#issuecomment-71470592>
> ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/switchonproject/sip-html5/issues/13#issuecomment-71644897>
> .
>



-- 
Esa Falkenroth
mobile: 070-666-14-14 (intl +46706661414)
work: 011-4958298 (intl +46114958298)
skype: Esa.Falkenroth
email: nospam@falkenroth.se (for business mail)
email: esa@falkenroth.se (for private mail)

**Comment by esafa** [comment]
Just to make sure we are on the same page: The focus is on user story number one. I know cismet has done some background work in parallel to prepare for future user stories. This is all good but we will implement the user story without additional functionality. 

Hence:
No INSPIRE keywords for now
No free-text search for now
No free keywords for now
No map for now
No country-selection for now

Here a quick capsule summary of the 30+ posts on our first user story.

Userstory is “As a hydrologist, I need to find a list of data links as provided in Emmas data table by hierarchical keywords as provided in CUAHSI's tabular ontology table in column B to D, so I can efficiently identify precise and required data for my study.
CUAHSI ontology: http://his.cuahsi.org/documents/Tabular_Ontology.xlsx 
Emma Metadata: already in sip

Clarifications by Rene:  “we meant Columns B (Nav1), C (Search1), D (Search2) “
Clarification by Pascal: “[need] tagging the available resources according to the CUAHSI ontology” 
Clarification by Sascha:  must add these keywords “explicitly and all the time” [..] to get the expected results.” [task for summer worker ]
Suggestion by Esa/Rene:  “some datasets to only have the coarse topic (column B) and skip the detail classification”
Suggestion by Sascha: “ first show all and then filter [with keywords] rather than show only what fits to the substring.
Clarification by Duro: “hydrologists are not aware of some cuashi list so fixed list is a good suggestion”
Clarification by Remko: “ scrolling through a list with keywords may be easier “

**Comment by schlob** [comment]
it seems that we will have the first working prototype today or tomorrow. Some aspects are not very stable yet:-) So we will post a vid before an actual release. The prototype is connected to the SIP Backend already so it works with the current (Emma) data.

**Comment by p-a-s-c-a-l** [comment]
Here is a demo video of the SIP HTML5 prototype: http://youtu.be/rBNTKHOfuVE
It shows how the client supports User Story 1.

**Comment by schlob** [comment]
so @duropa, @rcapell and @rcnijzink pls comment :-)

**Comment by esafa** [comment]
Yes comments please!  Very high tempo. I am duly impressed J



(1)    I hope @Rene, @Remko and @Duropa can give preliminary *feedback* on
the *functionality*.  I think you should be able to tell if the first
version matches what you asked for in the user story? @Sascha also made one
good suggestion (have a look if it is in).

(2)    The mantra for the HTML5-client is “easy to use for hydrologists”…
but *please wait with comments on usability/ease-of-use*. Just looking at a
video by a computer expert that knows the tool very well is really not a
good way reveal any problems with “ease of use”.

(3)    *Once we get the online link* to the tool , I want all (especially
Rene, Remko and Duro) to test the tool, comment on “easy use”, and see if
we can figure out “where do I click to pick the CUASHI keywords?”  and “How
do I show the download link(s)?”.  Suggestions for improvements are
welcome. But make sure any suggested improvements are really worth the time
and effort (every extra iteration on improvements means one iteration less
with new userstories.



Cheers,

//Esa

On Wed, Jan 28, 2015 at 4:25 PM, Sascha Schlobinski <
notifications@github.com> wrote:

> so @duropa <https://github.com/duropa>, @rcapell
> <https://github.com/rcapell> and @rcnijzink <https://github.com/rcnijzink>
> pls comment :-)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/switchonproject/sip-html5/issues/13#issuecomment-71852731>
> .
>



-- 
Esa Falkenroth
mobile: 070-666-14-14 (intl +46706661414)
work: 011-4958298 (intl +46114958298)
skype: Esa.Falkenroth
email: nospam@falkenroth.se (for business mail)
email: esa@falkenroth.se (for private mail)

**Comment by duropa** [comment]
yes, the video is nice, but would it be possible to test it by hand (online)?it is rather difficult for me to comment without practical experience...

**Comment by schlob** [comment]
ofc asap :-)

**Comment by schlob** [comment]
probably tomorrow

**Comment by rcnijzink** [comment]
This looks very good! I think the functionality with the list works like I had in mind. The list with results looks very structured, with not too much and not too little info ('to efficiently identify precise and required data'), and the actual download link looks easy to find as well. Of course, practical use will make it easier to comment, but at least my first impression is that it has the main functionalities we tried to cover with the user story.

**Comment by rcapell** [comment]
I agree with the above, this looks really nice, well done! 

At the risk of this being another user story:
One thing I would think could be good is an additional column "keyword", because the more generic keywords would yield several sub-categories and that would be useful to see in the results. E.g. keyword:(Pressure) yields (Pressure: Pressure, air; Pressure, water)

**Comment by esafa** [comment]
@rcapell :that would be a different userstory. I would like the userstory
group to prioritise  the next user story. if that keyword story is indeed
the next really most important part, by all means, put that up. but
first.we should have a telcon on the next story.
Den 28 jan 2015 17:51 skrev "rcapell" <notifications@github.com>:

**Comment by p-a-s-c-a-l** [comment]
HTML5 SIP client implementing User Story#1is online: http://www.water-switch-on.eu/sip-webclient/sip-alpha/

**Comment by ci-cismet-de** [comment]
Integrated branch 'feature/13-mockups-for-user-story-1 (476d0dcfeb6c370ef6bdd46c7e20ceedec112e1e)' into 'dev (1ff3b13208e18d6c2ccec462dcaa14faac731b54)' -- [Build 16](https://ci.cismet.de/job/sip-html5/16/)

**Comment by p-a-s-c-a-l** [comment]
@Rene, @Remko and @Duropa : Did you test the client? Any comments on that?
Can we close this user story?

**Comment by remko** [comment]
I'm afraid you're referencing the wrong Remko.

**Comment by rcnijzink** [comment]
I think it did what we wanted with regard to this user story. However, when I try to use it I cannot search or find, but I guess this is because it is not linked to actual (meta)-data, right?

**Comment by p-a-s-c-a-l** [comment]
> I think it did what we wanted with regard to this user story. However, when I try to use it I cannot search or find, but I guess this is because it is not linked to actual (meta)-data, right?

No, the client is linked to the meta-data repository. Which keyword did you try? keyword:"Area" should return one result. Does the client show an error message?

**Comment by rcnijzink** [comment]
I get: 'Search could no be perfomed:cannot search for resources'  when I use 'Flux:Precipitation'

**Comment by rcnijzink** [comment]
Same result for 'Area' btw

**Comment by p-a-s-c-a-l** [comment]
OK. I'm not able to reproduce this error. Perhaps a firewall or proxy issue or a problem with an outdated web browser. We have to check this out during the meeting. Gerben tested the client on Friday and it did work.

@esafa  Can you please test the search?

**Comment by rcnijzink** [comment]
I used internet explorer 10 on windows. Now, I turned to firefox on ubuntu and it worked. I get 1 result for area indeed. 'Flux:precipitation' is just not finding results, it gives 'Search completed, but no results found', is that possible?

**Comment by schlob** [comment]
yes the underlying data is not cuashi tagged yet (smhi summerworker)

**Comment by duropa** [comment]
dear pascal,
i have similar problem as remko...no results...i 
tried internet explorer 9 and also firefox (35.0)...

cheers
d.

At 09:48 2.2.2015, Pascal DihÃ© wrote:

><https://github.com/Rene>@Rene, 
><https://github.com/Remko>@Remko and 
><https://github.com/Duropa>@Duropa : Did you 
>test the client? Any comments on that?
>Can we close this user story?
>
>­
>Reply to this email directly or 
><https://github.com/switchonproject/sip-html5/issues/13#issuecomment-72422977>view 
>it on GitHub.

*******************************************************************************
Assoc. Prof. Juraj Parajka
Institute of Hydraulic Engineering and Water Resources Management
Vienna University of Technology
Karlsplatz 13/222, A-1040 Vienna, Austria
www.hydro.tuwien.ac.at
*******************************

**Comment by p-a-s-c-a-l** [comment]
**Internet Explorer >= 10 is currently not supported!** This issue will be addressed shortly in the [REST Service](https://github.com/cismet/cids-server-rest/issues/58). However, Firefox 35 should work. 

@duropa Do you get no results (That's o.k.see. Sascha's comment https://github.com/switchonproject/sip-html5/issues/13#issuecomment-72437122) or an error message in Fire Fox 35 ("cannot search for resources")?

**Comment by duropa** [comment]
now the firefox - after restart - works...ie 9.0 
not (it hangs in the middle of Please wait, search is in progress.)
d.

At 14:23 2.2.2015, you wrote:

>Internet Explorer >= 10 is currently not 
>supported! This issue will be addressed shortly 
>in the 
><https://github.com/cismet/cids-server-rest/issues/58>REST 
>Service. However, Firefox 35 should work.
>
><https://github.com/duropa>@duropa Do you get no 
>results (That's o.k.see. Sascha's comment 
><https://github.com/switchonproject/sip-html5/issues/13#issuecomment-72437122>#13 
>(comment)) or an error message in Fire Fox 35 ("cannot search for resources")?
>
>­
>Reply to this email directly or 
><https://github.com/switchonproject/sip-html5/issues/13#issuecomment-72456944>view 
>it on GitHub.

*******************************************************************************
Assoc. Prof. Juraj Parajka
Institute of Hydraulic Engineering and Water Resources Management
Vienna University of Technology
Karlsplatz 13/222, A-1040 Vienna, Austria
www.hydro.tuwien.ac.at
*******************************

**Comment by schlob** [comment]
<= I suppose

**Comment by esafa** [comment]
Yes.. I tested the search. Worked for me. I found results with keyword
"area" and some other as well. Running Win8.1 with latest desktop chrome.

My comments so far:

(1) The implementation seems to match the userstory quite well (but that is
really up to remko, rene and duro to judge).
(2) Obviously, I should have remembered to add the CUASHI-keywords to at
least a few data resources to make the testing more simple. I´ll see if I
can produce that with help of hydrologists here at SMHI.
(3) The userinteractions require quite many click.. I sent @p-a-s-c-a-l  a
link to ebays keywordsearch for comparison of user interactions... They
have implemented similar keyword search with only two clicks (ours have 7
clicks/drags).
(4) If you do the clicks in the wrong order it wont work (for instance when
scrolling the keywords the search button (looking glass) is blocked.
(5) Saschas idea of showing partial search results while the user adds
keywords is not implemented (but it should actually be a separate user
story). Ill put that one up as a candidate for fourth userstory (to avoid
feature creep)
(6) Still think the label "Search filter" is quite ambiguous. Consider
renaming it to something that gives the user a more clear idea what the
button does. Perhaps "Search by keyword"`?
(7) Performance is good.
(8) Seems stable too.

Good and fast work.

Thanks,
//Esa


On Mon, Feb 2, 2015 at 2:42 PM, Sascha Schlobinski <notifications@github.com
> wrote:

> <= I suppose
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/switchonproject/sip-html5/issues/13#issuecomment-72459336>
> .
>



-- 
Esa Falkenroth
mobile: 070-666-14-14 (intl +46706661414)
work: 011-4958298 (intl +46114958298)
skype: Esa.Falkenroth
email: nospam@falkenroth.se (for business mail)
email: esa@falkenroth.se (for private mail)

**Comment by p-a-s-c-a-l** [comment]
Thanks for the comments, @esafa!
Some answers below:

> (3) The userinteractions require quite many click.. I sent @p-a-s-c-a-l a link to ebays keywordsearch for comparison of user interactions... They have implemented similar keyword search with only two clicks (ours have 7 clicks/drags).

7 cklicks? I count 3: open search filter -> select keyword -> press search button.
Avoiding scrolling or doing a 3rd click is hardly possible: There are 72 CUASHI Keywords that need to be shown somehow in the ui. Even in ebay you have to scroll the list or do 3 clicks to see all categories: "Shop by category" -> "see all categories" -> "select category." Furthermore, ebay does not allow to search for more than one category, keyword seach in SWITCH-ON does. Anyway, we can do something similar like the ebay categories search. Just provide a user user story.

> (4) If you do the clicks in the wrong order it wont work (for instance when scrolling the keywords the search button (looking glass) is blocked.

Can you please provide more details here? What means wrong order? The search button is intentionally blocked, if the search string is either empty or invalid (format is `seachFilter:"parameter"`, e.g. geo:"POLYGON(...)" or keyword:"water").

> (5) Saschas idea of showing partial search results while the user adds keywords is not implemented (but it should actually be a separate user story). Ill put that one up as a candidate for fourth userstory (to avoid feature creep)

I don't think that showing partial search results is what Sascha meant. It's more like applying a post-search filter on search results. 

> (6) Still think the label "Search filter" is quite ambiguous. Consider renaming it to something that gives the user a more clear idea what the button does. Perhaps "Search by keyword"`?

O.K: Maybe we can find a better term than "Search Filter" But this ui control is not only used for keyword search but also to, e.g. show options for geospatial search as requests in  #25 (user story no. 2).

**Comment by duropa** [comment]
dear esa,
i'm quite happy with the implementation of the user story #1.
just two notes:
1) i do not like the starting page...the big: No 
data available, please perform a search... is not 
very nice, but probably is supposed to be changed in the final version

2) it is not clear, how the interface will be 
extended/further developed (by adding 
functionalities of next user stories- e.g. 
bounding box search, filtering, 
sorting)...probably this  is by purpose, but i 
will not be unhappy to see the direction a bit also now...

cheers
d.


At 03:19 3.2.2015, esafa wrote:
>Yes.. I tested the search. Worked for me. I found results with keyword
>"area" and some other as well. Running Win8.1 with latest desktop chrome.
>
>My comments so far:
>
>(1) The implementation seems to match the userstory quite well (but that is
>really up to remko, rene and duro to judge).
>(2) Obviously, I should have remembered to add the CUASHI-keywords to at
>least a few data resources to make the testing more simple. IÂ´ll see if I
>can produce that with help of hydrologists here at SMHI.
>(3) The userinteractions require quite many click.. I sent @p-a-s-c-a-l a
>link to ebays keywordsearch for comparison of user interactions... They
>have implemented similar keyword search with only two clicks (ours have 7
>clicks/drags).
>(4) If you do the clicks in the wrong order it wont work (for instance when
>scrolling the keywords the search button (looking glass) is blocked.
>(5) Saschas idea of showing partial search results while the user adds
>keywords is not implemented (but it should actually be a separate user
>story). Ill put that one up as a candidate for fourth userstory (to avoid
>feature creep)
>(6) Still think the label "Search filter" is quite ambiguous. Consider
>renaming it to something that gives the user a more clear idea what the
>button does. Perhaps "Search by keyword"`?
>(7) Performance is good.
>(8) Seems stable too.
>
>Good and fast work.
>
>Thanks,
>//Esa
>
>
>On Mon, Feb 2, 2015 at 2:42 PM, Sascha Schlobinski <notifications@github.com
> > wrote:
>
> > <= I suppose
> >
> > —
> > Reply to this email directly or view it on GitHub
> > 
> <https://github.com/switchonproject/sip-html5/issues/13#issuecomment-72459336>
> > .
> >
>
>
>
>--
>Esa Falkenroth
>mobile: 070-666-14-14 (intl +46706661414)
>work: 011-4958298 (intl +46114958298)
>skype: Esa.Falkenroth
>email: nospam@falkenroth.se (for business mail)
>email: esa@falkenroth.se (for private mail)
>
>­
>Reply to this email directly or 
><https://github.com/switchonproject/sip-html5/issues/13#issuecomment-72580365>view 
>it on GitHub.

*******************************************************************************
Assoc. Prof. Juraj Parajka
Institute of Hydraulic Engineering and Water Resources Management
Vienna University of Technology
Karlsplatz 13/222, A-1040 Vienna, Austria
www.hydro.tuwien.ac.at
*******************************

**Comment by schlob** [comment]
OK - I think the general functionality is fulfilled. OFC there can be enhancements any time. If they are important enough - let's define a new User Story.

Following the User Story approach we would need something on an integrated UI.

As hydroguy I want seamless integration of spatial and keyword search to find data that matches my keywords in a certain area. So I can better identify the candidate data for my experiment.

OFC we have a proposal already that we will not push here without user story because Esa would
beat us on the head;-)

**Comment by ci-cismet-de** [comment]
Integrated branch 'feature/13-mockups-for-user-story-1 (29e6da89a5a613c1c9ead519cb0fbdf67a4dae78)' into 'dev (23967d5a7a0404e1ebc5e10c3ed88e4ba8b3970e)' -- [Build 25](https://ci.cismet.de/job/sip-html5/25/)

**Comment by esafa** [comment]
>>> 7 cklicks? I count 3: open search filter -> select keyword -> press
search button.
>>> [..] Ebay you have to scroll the list or do 3 clicks to see all
categories: "Shop by category" -> "see all categories" -> "select
category." [..] Anyway, we can do something similar like the ebay
categories search. Just provide a user user story.

You are right Pascal. The number of clicks/moves/drags are very comparable
with Ebay that should be "top of the line" in ease-of-use.

Here is a more accurate mouse and keyboard interaction count for user story
#1 Finding "detailed keyword" such "Level, lake"...

<------------------------- view in wide window
------------------------------------------------------------->

*HTML5-client keyword search:                **Ebay keyword search:*

- move (to "Search filter"),                - move (shop by category),
- click (open menu),                        - click (open menu),
                                            - move(See all categories),
 ("Ebay has two levels of keywords")
                                            - click,
- move(to popup scrollbar),                 - move(to normal scrollbar),
- drag(to find correct keyword),            - drag(to find correct
keyword),
- move (to e.g. "Level, lake"),             - move (to e.g. "Classic
cars"),
- click(puts keyword into the textbox),     - click(gives results),
 ("Ebay only has one selection")
- move(to looking glass),
- click (to search),
- move (to scrollbar for multiple matches), - move(to scrollbar),
- drag(to scroll search matches),           - drag(to scroll search
matches).
- move (to "show download link"),           - move (to some car entry)
- click (to open download popup)            - click

(6 moves, 4 clicks, 2 drags)                (6 moves, 4 clicks, 2 drags)

Gratz :)

While doing my first count of interactions, I could not click on the
"looking glass" after selecting keyword. The pulldown menu deactivated
clicking on the "looking glass" button, so I had to first close the menu
(one click) before doing the search ( just like you did in the mockup
video). That issues seems to be resolved so now the number of
clicks/drags/moves are good.

Cheers,
//Esa














On Tue, Feb 3, 2015 at 11:33 AM, Sascha Schlobinski <
notifications@github.com> wrote:

> Closed #13 <https://github.com/switchonproject/sip-html5/issues/13>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/switchonproject/sip-html5/issues/13#event-229158045>.
>



-- 
Esa Falkenroth
mobile: 070-666-14-14 (intl +46706661414)
work: 011-4958298 (intl +46114958298)
skype: Esa.Falkenroth
email: nospam@falkenroth.se (for business mail)
email: esa@falkenroth.se (for private mail)

**Comment by rcapell** [comment]
one comment on what i think is a usability bug:
The google-like search field suggests that users can type search terms by hand and get results, whereas only specifically formatted searches are allowed (filter:"existing_keyword"). This is misleading imho.

**Comment by rcapell** [comment]
The description text in results indicates that there is more text in the queried database but there is no possibility to read it, e.g. by clicking on the '[...]' indicator.
Would that be possible to change easily?

**Comment by p-a-s-c-a-l** [comment]
> The google-like search field suggests that users can type search terms by hand and get results, whereas only specifically formatted searches are allowed (filter:"existing_keyword"). This is misleading imho.

yes, that's probably misleading. We'll try to improve this without breaking the functionality to define the search filter directly in this box, e.g. by directly entering bbox coordinates with a geo:"POYGON(...)" statement. 

> The description text in results indicates that there is more text in the queried database but there is no possibility to read it, e.g. by clicking on the '[...]' indicator.
> Would that be possible to change easily?

yes, that's no problem.

**Comment by esafa** [comment]
CUASHI column D item "Flux, DIscharge" is missing in the pulldown list of the new demo (demo laptop) shown in Saarbrucken.

**Comment by esafa** [comment]
Seems the column D data is only visible in one of the two keyword search options. Cascading menu might be better than two separate filters.

**Comment by p-a-s-c-a-l** [comment]
> CUASHI column D item "Flux, DIscharge" is missing in the pulldown list of the new demo (demo laptop) shown in Saarbrucken.

No, we accidentally looked for "discharge" in the table (scrolling down to 'd...') but the correct CUASHI Term is _Flux, Discharge_.

> Seems the column D data is only visible in one of the two keyword search options. Cascading menu might be better than two separate filters.

O.K.
